src/main/java/duke/utilities/Parser.java

```diff
@@ -0,0 +1,252 @@
+package duke.utilities;
+
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+import duke.exceptions.ContentEmpty;
+import duke.exceptions.DateParseException;
+import duke.exceptions.DukeException;
+import duke.exceptions.IncompleteCommandException;
+import duke.exceptions.InvalidMarkInput;
+import duke.tasks.ITask;
+
+
+/**
+ * Parser class to translate user input to valid program input
+ */
+public class Parser {
+    /**
+     * Pre-define input date format
+     */
+    public static final DateFormat DATE_IN_FORMAT = new SimpleDateFormat("dd/MM/yyyy HHmm");
+
+    /**
+     * Pre-define output date format
+     */
+    public static final DateFormat DATE_OUTPUT_FORMAT = new SimpleDateFormat("MM-dd-yyyy HH:mm:ss aa");
+
+
+    private boolean processed = false;
+    private String input;
+    private int index;
+    private String description;
+    private ITask.TaskTypes type;
+    private Date by;
+    private Date from;
+    private Date to;
+
+    private TaskManager taskManager;
+
+    /**
+     * Constructor for Parser
+     *
+     * @param input input from user
+     */
+    public Parser(String input) {
+        this.input = input;
+    }
+
+    /**
+     * Constructor for Parser
+     *
+     * @param taskManager to handle to task
+     */
+    public Parser(TaskManager taskManager) {
+        this.taskManager = taskManager;
+    }
+
+    /**
+     * Constructor for Parser
+     *
+     * @param input       input from user
+     * @param taskManager to handle to task
+     */
+    public Parser(String input, TaskManager taskManager) {
+        this.input = input;
+        this.taskManager = taskManager;
+    }
+
+    /**
+     * Constructor for Parser
+     *
+     * @param input       input from user
+     * @param taskManager to handle to task
+     * @param type        of the command
+     */
+    public Parser(String input, TaskManager taskManager, ITask.TaskTypes type) {
+        this.taskManager = taskManager;
+        this.input = input;
+        this.type = type;
+    }
+
+
+    /**
+     * Return the 'by' date of deadline task
+     *
+     * @throws DukeException IF error occur
+     */
+    public Date getBy() throws DukeException {
+        if (by == null) {
+            throw new ContentEmpty("'by'");
+        }
+        return by;
+
+    }
+
+
+    /**
+     * Convert the user input to valid program input for a deadline
```

Comment:
I noticed this elsewhere too, your Javadoc comments should begin with a singular verb instead of a plural one.